backend/editor/api.py

```diff
@@ -41,16 +44,26 @@
 @app.on_event("startup")
 async def startup():
     """
-    Initialize database
+    Initialize Neo4J database
     """
-    initialize_db()
+    global driver, session
+    uri = settings.uri
+    driver = GraphDatabase.driver(uri)
 
 @app.on_event("shutdown")
 async def shutdown():
     """
-    Shutdown database
+    Close driver of Neo4J database
     """
-    shutdown_db()
+    driver.close()
+
+@app.middleware("http")
+async def transaction_creation(request: Request, call_next):
+    with driver.session() as session:
+        global txn
+        txn = session.begin_transaction()
+        response = await call_next(request)
+        return response

```

Comment:
Put it in it's own file (with initialize_db and shutdown_db. So that you don't have cross import problems.

Maybe graph_db.py